src/hooks/libp2p/useLibp2p.ts

```diff
@@ -55,7 +55,7 @@ export function useLibp2p() {
   const onPeerConnect = useCallback(
     (evt: CustomEvent<Connection>) => {
       const peerId = evt.detail.remotePeer.toString();
-      dispatch(setArchaeologistConnection(peerId, evt.detail));
+      setTimeout(() => dispatch(setArchaeologistConnection(peerId, evt.detail)), 500);
```

Comment:
Added this small delay because public keys are also being sent over and set here in the webapp -- leading to a race condition where set public key uses the "old" selected archaeologist list that had an undefined connection, right after this set connection has been dispatched, and overwriting the newly (re)set connection.

Note we have not really been depending on this listener to set node connections -- that happens directly in `useDialArchaeologists`, and that is what is overwritten, I think. But this is now important for the edge case where the node drops connection and dial has to be called again.

Reply:
I'm wondering if this `setArchaeologistConnection` could be removed. The connections get set in the `dial` routine, so is this necessary?

Reply:
Ah yeah you're right. The delay can be moved to the dial routine with same effect

Reply:
This is removed in #218, which branches of this PR.